### PR TITLE
Chore: Bump stale bot version and exclude certain cases

### PR DESCRIPTION
**Pull request by joaopgrassi**
This bumps the version of the GH action, excludes draft PRs and PRs with the following labels on them:

- `bug`
- `work in progress` (New tag to be created)
- `experts needed`

Please let me know if you think these labels are good or if you have any other ideas.

### CODE REVIEW

**Comment by joaopgrassi** [comment]
I opened a [bug](https://github.com/actions/stale/issues/1132) with them, because I can't understand why this pr #627 still has the `Stale` label, even after receiving updates.

File: .github/workflows/stale-pr.yml [hunk 1]
```diff
@@ -7,11 +7,12 @@ jobs:
   stale:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/stale@v3
+      - uses: actions/stale@v9
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-          stale-pr-message: 'This PR was marked stale due to lack of activity. It will be closed in 7 days.'
+          stale-pr-message: 'This PR was marked stale due to lack of activity. It will be closed in 7 days. Consider applying the label `work in progress` to avoid this being marked as stale again.'
```
Comment (arminru) [inline .github/workflows/stale-pr.yml:13]: Non-approvers won't be able to add this label - which makes sense but the message might therefore confuse PR authors.

File: .github/workflows/stale-pr.yml [hunk 2]
```diff
@@ -7,11 +7,12 @@ jobs:
...
           close-pr-message: 'Closed as inactive. Feel free to reopen if this PR is still being worked on.'
-          exempt-pr-labels: 'release:after-ga'
+          exempt-pr-labels: 'bug,work in progress,experts needed'
```
Comment (arminru) [inline .github/workflows/stale-pr.yml:15]: Would this need quoting/escaping for the spaces?

**Review by arminru — APPROVED** [review]

File: .github/workflows/stale-pr.yml (hunk 2 above)
Comment (joaopgrassi) [reply .github/workflows/stale-pr.yml:15]: Doesn't seem so by looking at examples in other repos

Comment (joaopgrassi) [reply .github/workflows/stale-pr.yml:15]: It says a "comma separated list" so I think that should work. https://github.com/actions/stale?tab=readme-ov-file#exempt-pr-labels

File: .github/workflows/stale-pr.yml (hunk 1 above)
Comment (joaopgrassi) [reply .github/workflows/stale-pr.yml]: True.. I removed it for now.

**Review by trisch-me — APPROVED** [review]

**Comment by trisch-me** [comment]
@joaopgrassi as non-approvers can't update the labels maybe we should provide another option for them. 

There are apps like [multi-labeler](https://github.com/fuxingloh/multi-labeler#features) where we could set up a regexp for title to create a label based on it. In this case we could add `work in progress` for `WIP` in title as an example.

**Review by jsuereth — APPROVED** [review]

**Pull request merged**